src/ServiceControl.Monitoring/QueueLength/QueueLengthDataStore.cs

```diff
@@ -63,16 +63,18 @@ static MonitoredEndpointValues AggregateQueueLength(List<EndpointInstanceInterva
         {
             Func<long, double> returnOneIfZero = x => x == 0 ? 1 : x;
 
-            return new MonitoredEndpointValues
+            var result = new MonitoredEndpointValues
             {
-                Average = intervals.Sum(t => t.TotalValue) / returnOneIfZero(intervals.Sum(t => t.Intervals.Length)),
                 Points = intervals.SelectMany(t => t.Intervals)
                     .GroupBy(i => i.IntervalStart)
                     .OrderBy(g => g.Key)
                     .Select(g => g.Sum(i => i.TotalValue / returnOneIfZero(i.TotalMeasurements)))
                     .ToArray()
             };
 
+            result.Average = result.Points.Length > 0 ? result.Points.Average() : 0d;
```

Comment:
I see a fix applied on a commit that was green on the build server without a test being changed? Are we lacking the coverage in here?